R/GameState.R

```diff
@@ -0,0 +1,87 @@
+#' R6 Class representing a game state
+#'
+#' A game state has a player, list of ghosts and an arena.
+GameState <- R6::R6Class("GameState", #nolint
+  private = list(
+    .player = NULL,
+    .ghosts = NULL,
+    .arena = NULL,
+    check = function(player, ghosts, arena) {
```

Comment:
I feel the name `check` is a bit too general, as it does not tell you what it checks. But I understand the idea and thinkit is fine!

However, in this case, it checks the three input arguments. I would have preferred three functions, one per argument. I do think this is good enough, so let's just keep it as is!